opentelemetry-api/src/opentelemetry/attributes/__init__.py

```diff
@@ -173,19 +278,24 @@ def __setitem__(self, key: str, value: types.AttributeValue) -> None:
                 self.dropped += 1
                 return
 
-            value = _clean_attribute(key, value, self.max_value_len)  # type: ignore
-            if value is not None:
-                if key in self._dict:
-                    del self._dict[key]
-                elif (
-                    self.maxlen is not None and len(self._dict) == self.maxlen
-                ):
-                    if not isinstance(self._dict, OrderedDict):
-                        self._dict = OrderedDict(self._dict)
-                    self._dict.popitem(last=False)  # type: ignore
-                    self.dropped += 1
-
-                self._dict[key] = value  # type: ignore
+            if self._extended_attributes:
```

Comment:
Not a blocking comment, but I'm wondering if it would be preferable and feasible to define something like a `BoundedExtendedAttributes` type rather having `BoundedAttributes` have two different modes of operation depending on a constructor flag.

Reply:
I was going to make the same comment or maybe suggest having a subclass. As it is, we have to open up the allowed type on these methods to `types.AnyValue` which is unfortunate. Ideally we should have a generic for `MutableMapping[T]` on L231 but IIRC it was infeasible with mypy when we first tried.

I don't want to block but we should at least add a todo IMO